src/components/ApplicationGroup/AppGroupAppFilter.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { useState } from 'react'
+import ReactSelect, { components, MultiValue } from 'react-select'
+import { ReactComponent as ShowIcon } from '../../assets/icons/ic-visibility-on.svg'
+import { ReactComponent as ShowIconFilter } from '../../assets/icons/ic-visibility-on-filter.svg'
+import { ReactComponent as Search } from '../../assets/icons/ic-search.svg'
+import { ReactComponent as Clear } from '../../assets/icons/ic-error.svg'
+import { OptionType } from '../app/types'
+import { useAppGroupAppFilterContext } from './AppGroupDetailsRoute'
+import { appGroupAppSelectorStyle, getOptionBGClass } from './AppGroup.utils'
+
+const ValueContainer = (props) => {
+    let selectedAppsLength = props.getValue().length
+
+    const selectorText = `${selectedAppsLength > 0 ? selectedAppsLength : props.options.length}/${
+        props.options.length
+    } Applications`
+    return (
+        <components.ValueContainer {...props}>
+            {!props.selectProps.inputValue ? (
+                <>
+                    {!props.selectProps.menuIsOpen ? (
+                        <>
+                            {selectedAppsLength > 0 ? (
+                                <ShowIconFilter className="icon-dim-16 mr-4 mw-18" />
+                            ) : (
+                                <ShowIcon className="icon-dim-16 mr-4 mw-18" />
+                            )}
+                            <span className="dc__position-abs dc__left-35 cn-9 ml-2">{selectorText}</span>
+                        </>
+                    ) : (
+                        <>
+                            <Search className="icon-dim-16 mr-4 mw-18" />
+                            <span className="dc__position-abs dc__left-35 cn-5 ml-2">
+                                {props.selectProps.placeholder}
+                            </span>
+                        </>
+                    )}
+                </>
+            ) : (
+                <Search className="icon-dim-16 mr-4 mw-18" />
+            )}
+            {React.cloneElement(props.children[1])}
+        </components.ValueContainer>
+    )
+}
+
+const Option = (props) => {
```

Comment:
Can't we import it from react select utils?

Reply:
No. The use case is specific